syntax = "proto3";

package content_management;

service ContentManagement {
  rpc CreateTextContent (CreateTextContentRequest) returns (CreateTextContentResponse);
}

service UserManagement {
  rpc CreateUser (CreateUserRequest) returns (CreateUserResponse);
}

message CreateTextContentRequest {
  string title = 1;
  string content = 2;
  int32 author_id = 3;
}

message CreateTextContentResponse {
  int32 content_id = 1;
  string message = 2;
}

message CreateUserRequest {
  string username = 1;
  string email = 2;
  string password = 3;
}

message CreateUserResponse {
  int32 user_id = 1;
  string message = 2;
}
